src/main/java/org/broadinstitute/hellbender/utils/Utils.java

```diff
@@ -589,6 +605,25 @@ public static void containsNoNull(final Collection<?> c, final String message) {
     }
 
     /**
+     * Checks that the collection does not contain a duplicate value (throws an {@link IllegalArgumentException} if it does).
+     * The implementation creates a {@link Set} as an intermediate step or detecting duplicates and returns this Set because
+     * it is sometimes useful to do so.
+     *
+     * @param c collection
+     * @param message A message to emit in case of error, in addition to reporting the first duplicate value found.
+     * @throws IllegalArgumentException if a {@code o == null}
+     */
+    public static <E> Set<E> containsNoDuplicates(final Collection<E> c, final String message) {
```

Comment:
add unit tests for containsNoDuplicates. Also, it's a strange name - does not suggest that it will blow up in my face or return a set, rather the name suggests that it will return a boolean and not blow up.